crates/iota-indexer/src/indexer_reader.rs

```diff
@@ -1568,6 +1568,17 @@ impl IndexerReader {
             diesel::sql_query("SELECT * FROM network_metrics;")
                 .get_result::<StoredNetworkMetrics>(conn)
         })?;
+        if metrics.tps_30_days == -1.0 {
+            // this implies that the value is not available in the db
+            metrics.tps_30_days = 0.0;
+        }
```

Comment:
Why did you opt out from `COALESCE(..., 0)`?

Reply:
As the `estimates` for the other relations can return `-1` I thought it would be cleaner to handle it also for the `tps_30_days` value in a similar way. Also I guess `-1` better reflects that the value is not available yet.

Reply:
It is not cleaner though. Returning `-1` for the other values relying on [estimates](https://www.postgresql.org/docs/current/row-estimation-examples.html) is well defined and consistent with postgres docs.

Returning `-1.0` for an actual value is an ad-hoc convention that is not transparent in my opinion.

Reply:
I see and agree it's not the optimal. However, also I think returning `0.0` for 30d TPS is not really clean.
But since it's an actual value as you mentioned, maybe it's better to rely on `0.0`. I will change.

Reply:
Fixed with https://github.com/iotaledger/iota/pull/5632/commits/6f1da5844aec5584a189ed490ba0861b57113873